api/test/nostd/string_view_test.cc

```diff
@@ -71,8 +71,8 @@ TEST(StringViewTest, SubstrPortion)
 TEST(StringViewTest, SubstrOutOfRange)
 {
   string_view s = "abc123";
-#if __EXCEPTIONS || ((defined(OPENTELEMETRY_STL_VERSION) && (OPENTELEMETRY_STL_VERSION >= 2020)))
-  EXPECT_THROW(s.substr(10), std::out_of_range);
+#if __EXCEPTIONS || (defined(OPENTELEMETRY_STL_VERSION) && (OPENTELEMETRY_STL_VERSION >= 2020))
+  EXPECT_THROW((void)s.substr(10), std::out_of_range);
```

Comment:
I suspect the existing code was incorrect, as std::string_view is introduced in C++17, not C++20.

Please investigate (local CXX 17 MSVC build) if the logic should be `OPENTELEMETRY_STL_VERSION >= 2017` instead.

Reply:
Verified that std::out_of_range exception was thrown out with MSVC/C++17. Thanks for the catch.